src/test/scala/pirate.example/GitExample.scala

```diff
@@ -15,42 +17,52 @@ object GitHtmlPath extends GitCommand
 object GitManPath extends GitCommand
 object GitInfoPath extends GitCommand
 case class GitHelp(command: Option[String]) extends GitCommand
+case class GitAdd(force: Boolean, interactive: Boolean, patch: Boolean, edit: Boolean, pathspec: List[File]) extends GitCommand
+case class GitRm(force: Boolean, dryRun: Boolean, recurse: Boolean, cached: Boolean, pathspec: List[File]) extends GitCommand
 
 object GitMain extends PirateMainIO[Git] {
   val version: Parse[GitCommand] =
-    terminator("version", GitVersion)
+    terminator(long("version") |+| short('v') |+| description("Prints the Git suite version that the git program came from."), GitVersion)
 
   val help: Parse[GitCommand] =
-    terminatorx("help", GitHelp.apply)
+    terminatorx(long("help") |+| description("Prints the synopsis and a list of the most commonly used commands. If a Git command is named this option will bring up the manual page for that command."), GitHelp.apply)
 
-  val cwd: Parse[String] =
-    flag('C', "<path>")
+  val cwd: Parse[Option[String]] =
+    flag[String](short('C') |+| metavar("<path>") |+| description("Run as if git was started in <path> instead of the current working directory.")).option
 
-  val conf: Parse[String] =
-    flag('c', "<name>=<value>")
+  val conf: Parse[Option[String]] =
+    flag[String](short('c') |+| metavar("<name>=<value>") |+| description("Pass a configuration parameter to the command.")).option
 
-  val exec: Parse[String] =
-    flag("exec-path", "<path>") // FIX fork on arg terminator vs option
+  val exec: Parse[Option[String]] =
+    flag[String](long("exec-path") |+| metavar("<path>") |+| description("Path to wherever your core Git programs are installed.")).option
 
   val html: Parse[GitCommand] =
-    terminator("html-path", GitHtmlPath)
+    terminator(long("html-path") |+| description("Print the path, without trailing slash, where Git's HTML documentation is installed and exit."), GitHtmlPath)
 
   val man: Parse[GitCommand] =
-    terminator("man-path", GitManPath)
+    terminator(long("man-path") |+| description("Print the manpath (see man(1)) for the man pages for this version of Git and exit."), GitManPath)
 
   val info: Parse[GitCommand] =
-    terminator("info-path", GitInfoPath)
+    terminator(long("info-path"), GitInfoPath)
+
+  // Applicitive style GitAdd |*| (switch ...) leads to invariance issues.
```

Comment:
If the lines are replaced with GitAdd |*| (switch(long("force")), switch, ...))
then the compiler complains with 
[error] /.../pirate.example/GitExample.scala:49: type mismatch;
[error]  found   : pirate.Parse[pirate.example.GitAdd]
[error]  required: pirate.Parse[pirate.example.GitCommand]
[error] Note: pirate.example.GitAdd <: pirate.example.GitCommand, but trait Parse is invariant in type A.
[error] You may wish to define A as +A instead. (SLS 4.5)